packages/core-data/src/hooks/use-entity-records.ts

```diff
@@ -77,9 +84,15 @@ export default function __experimentalUseEntityRecords< RecordType >(
 	const queryAsString = addQueryArgs( '', queryArgs );
 
 	const { data: records, ...rest } = useQuerySelect(
-		( query ) =>
-			query( coreStore ).getEntityRecords( kind, name, queryArgs ),
-		[ kind, name, queryAsString ]
+		( query ) => {
+			if ( ! options.enabled ) {
+				return {
+					data: [],
```

Comment:
This probably should const `EMPTY_ARRAY`, so a new reference isn't returned each time.

Reply:
Good spot! Let's tackle it as a part of making this a public API